src/Sparrow.Server/Platform/Win32/Win32MemoryMethods.cs

```diff
@@ -7,19 +7,161 @@
 using Sparrow.Json.Parsing;
 using Sparrow.Utils;
 
-// ReSharper disable InconsistentNaming
-
 namespace Sparrow.Server.Platform.Win32
 {
-    public static unsafe class Win32MemoryQueryMethods
+    public static unsafe class Win32MemoryMethods
     {
+        [return: MarshalAs(UnmanagedType.Bool)]
+        [DllImport("kernel32.dll", SetLastError = true)]
+        public static extern bool GetPhysicallyInstalledSystemMemory(out long totalMemoryInKb);
+
+        [return: MarshalAs(UnmanagedType.Bool)]
+        [DllImport("kernel32.dll", SetLastError = true)]
+        public static extern bool GlobalMemoryStatusEx(MemoryStatusEx* lpBuffer);
+
+        [DllImport("psapi.dll", SetLastError = true)]
+        public static extern bool GetProcessMemoryInfo(IntPtr hProcess, out PROCESS_MEMORY_COUNTERS_EX2 counters, uint size);
+
+        [DllImport("psapi.dll", SetLastError = true, EntryPoint = "GetProcessMemoryInfo")]
+        public static extern bool GetProcessMemoryInfoLegacy(IntPtr hProcess, out PROCESS_MEMORY_COUNTERS_EX counters, uint size);
```

Comment:
The difference between those is that this is a change in the _size_ of the struct you pass?

Reply:
Correct